homeassistant/components/powerwall/switch.py

```diff
@@ -0,0 +1,62 @@
+"""Support for Powerwall Switches (V2 API only)."""
+
+from typing import Any
+
+from tesla_powerwall import GridStatus, IslandMode
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import PowerWallEntity
+from .models import PowerwallRuntimeData
+
+ICON = "mdi:home-battery-outline"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Powerwall switch platform from Powerwall resources."""
+    powerwall_data: PowerwallRuntimeData = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([PowerwallOffGridEnabledEntity(powerwall_data)])
+
+
+class PowerwallOffGridEnabledEntity(PowerWallEntity, SwitchEntity):
+    """Representation of a Switch entity for Powerwall Off-grid operation."""
+
+    _attr_name = "Take Powerwall Off-Grid"
+    _attr_unique_id = "take_powerwall_off_grid"
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_device_class = SwitchDeviceClass.SWITCH
+    _attr_icon = ICON
```

Comment:
https://developers.home-assistant.io/docs/core/entity?_highlight=entity#advanced-properties
Icon to use in the frontend. Icons start with mdi: plus an [identifier](https://materialdesignicons.com/). You probably don't need this since Home Assistant already provides default icons for all entities according to its device_class. **This should be used only in the case where there either is no matching device_class or where the icon used for the device_class would be confusing or misleading.**

```suggestion
```